### PR TITLE
Feature/note selection request handler

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

File: src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs [hunk 1]
```diff
@@ -571,7 +574,123 @@ public async Task GetNoteCountByStatusAndAatf_GivenStatusAndAatf_ShouldReturnCor
...
+                note1ToBeFound.NoteTonnage.Add(new NoteTonnage(WeeeCategory.ConsumerEquipment, 1, null));
+                note1ToBeFound.NoteTonnage.Add(new NoteTonnage(WeeeCategory.GasDischargeLampsAndLedLightSources, 2, null));
+
+                context.Notes.Add(note1ToBeFound);
+
+                // note not to be found null category, matching scheme and status
+                var note2ToNotBeFound = await SetupSingleNote(context, database, NoteType.EvidenceNote, scheme);
+                note2ToNotBeFound.UpdateStatus(NoteStatus.Submitted, context.GetCurrentUser());
+                note2ToNotBeFound.UpdateStatus(NoteStatus.Approved, context.GetCurrentUser());
+                note2ToNotBeFound.NoteTonnage.Add(new NoteTonnage(WeeeCategory.ConsumerEquipment, null, null));
```
Comment (meglenapnovoselska) [inline src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs:601]: it's matching the category accepted but it's not matching the tonnage required so maybe a bit more specific on the comment, not important at all

File: src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs [hunk 2]
```diff
@@ -571,7 +574,123 @@ public async Task GetNoteCountByStatusAndAatf_GivenStatusAndAatf_ShouldReturnCor
             }
         }
 
-        private async Task<Note> SetupSingleNote(WeeeContext context, DatabaseWrapper database, NoteType noteType = null)
+        [Fact]
+        public async Task GetNotesToTransfer_GivenSchemeAndCategories_NotesShouldBeReturned()
```
Comment (meglenapnovoselska) [inline src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs:578]: would you add another note where the tonnage received is null but reused is not null?

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs [hunk 3]
```diff
@@ -286,6 +319,227 @@ public async Task TransferEvidenceNotePost_GivenInvalidModel_SessionShouldNotBeE
...
+
+            A.CallTo(() => mapper.Map<TransferEvidenceNotesViewModelMapTransfer, TransferEvidenceNotesViewModel>(
+                A<TransferEvidenceNotesViewModelMapTransfer>.That.Matches(t =>
+                    t.OrganisationId.Equals(organisationId) &&
+                    t.Notes.Equals(notes) &&
+                    t.Request.Equals(request)))).MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task TransferFromGet_GivenTransferNoteSessionObjectAndNotes_ModelShouldBeReturned()
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs:411]: update the name to WithValidModel or something, i think it's copy pasted from above,

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs [hunk 4]
```diff
@@ -286,6 +319,227 @@ public async Task TransferEvidenceNotePost_GivenInvalidModel_SessionShouldNotBeE
...
+
+            // act
+            var result = await transferEvidenceController.TransferFrom(organisationId) as ViewResult;
+
+            // assert
+            result.Model.Should().Be(model);
+        }
+
+        [Fact]
+        public async Task TransferFromGet_GivenTransferNoteSessionObjectAndNotes_TransferFromViewShouldBeReturned()
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs:427]: does it need to be updated?

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/ViewModels/TransferEvidenceNotesViewModelTests.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            result.SelectedEvidenceNotePairs.Add(new GenericControlPair<Guid, bool>(Guid.NewGuid(), true));
+            result.SelectedEvidenceNotePairs.Add(new GenericControlPair<Guid, bool>(Guid.NewGuid(), true));
+            result.SelectedEvidenceNotePairs.Add(new GenericControlPair<Guid, bool>(Guid.NewGuid(), false));
+
+            //act
+            var validationResults = result.Validate(new ValidationContext(result));
+
+            //assert
+            validationResults.Should().Contain(v =>
+                v.ErrorMessage.Equals("You cannot select more than 5 notes")
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Scheme/ViewModels/TransferEvidenceNotesViewModelTests.cs:66]: I cannot see that message in the CC, only can find "Select up to 5 notes", I guess it's different

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/TransferEvidenceNotesViewModelMapTransfer.cs [hunk 6]
```diff
@@ -3,23 +3,28 @@
...
 
         public Guid OrganisationId { get; private set; }
 
-        public TransferEvidenceNotesViewModelMapTransfer(List<EvidenceNoteData> notes,
+        public TransferEvidenceNotesViewModelMapTransfer(IList<EvidenceNoteData> notes,
             TransferEvidenceNoteRequest transferRequest,
-            Guid organisation)
+            Guid organisationId)
         {
+            Condition.Requires(notes).IsNotNull();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/TransferEvidenceNotesViewModelMapTransfer.cs:21]: Can you add IsNotEmpty() too?

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs:601]: Updated description.

File: src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.DataAccess.Tests.Integration/NotesDataAccessIntegration.cs:578]: Good idea, have added

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs (hunk 3 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs]: Updated

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs (hunk 4 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs]: Updated

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/ViewModels/TransferEvidenceNotesViewModelTests.cs (hunk 5 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/Areas/Scheme/ViewModels/TransferEvidenceNotesViewModelTests.cs:66]: Have left this as can be empty at the moment as in theory the user could filter out all the results when we get to that bit.

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/TransferEvidenceNotesViewModelMapTransfer.cs (hunk 6 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/TransferEvidenceNotesViewModelMapTransfer.cs:21]: Have left this as can be empty at the moment as in theory the user could filter out all the results when we get to that bit.

Comment (meglenapnovoselska) [reply src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/TransferEvidenceNotesViewModelMapTransfer.cs:21]: okay

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/ViewModels/TransferEvidenceNotesViewModelTests.cs (hunk 5 above)
Comment (meglenapnovoselska) [reply src/EA.Weee.Web.Tests.Unit/Areas/Scheme/ViewModels/TransferEvidenceNotesViewModelTests.cs:66]: okay

**Pull request merged**